### ISSUE TITLE
Refactor `CustomDiffFields` [5]

### ISSUE THREAD

**Issue by Sepehr-Sobhani** [bcgov/cas-cif#1391]
###Description of the Tech Debt###

Based on @pbastia  comment: https://github.com/bcgov/cas-cif/pull/1294#discussion_r1085983684
We need to maintain separately an `AmendmentSpecificCustomDiffFields`, and load it where appropriate instead. It would improve readability and make the classes smaller. We could define an interface for it too.

###Tech Debt Triage#
The purpose of our technical debt triage process is to analyze technical debt to determine risk level of the technical debt and the value in tackling that technical debt.

Risk Value Scoring:

| Level  | Value |
| ------ | ----- |
| High   | 3     |
| Medium | 2     |
| Low    | 1     |

| Technical Debt - Risk Types                                                                                                   | Level | Value |
| ----------------------------------------------------------------------------------------------------------------------------- | ----- | ----- |
| Business Area Risk - Risk of business area visibility / damage to user experience                                             |       |    1   |
| Developer Fault Risk - How likely will this tech debt cause a future error related to coding on top of it                     |       |      1 |
| System Fault Risk - Risk of system errors or application downtime                                                             |       |    1   |
| Time Scale Risk - Compound risk effect if left alone. How much more difficult to fix or dangerous will this become over time? |       |    1   |
| Time Sink Risk - How much will this tech debt slow the development process down                                               |       |    1   |
| TOTAL SCORE:                                                                                                                  |       |    5   |

**Comment by Sepehr-Sobhani** [comment]
@BCerki Yes you are right 👍 but I think it would be better to still have this as a placeholder and remove it later once the new approach is in place. What do you think?!

**Comment by BCerki** [comment]
Yes, we can leave this card, just wanted to flag that it was related to wild wild west
